### ISSUE TITLE
forge install/remove should automatically update remappings.txt and foundry.toml

### ISSUE THREAD

**Issue by StErMi** [gakonst/foundry#883]
### Component

Forge

### Describe the feature you would like

I would expect forge install/remove command to automatically update `remapping.txt` if present and also `foundry.toml` if inside the configuration file I've declared the remapping.

**Comment by mattsse** [comment]
good point, 
I think this is possible but will need to test this for toml, which is a bit more complex than remappings.txt

**Comment by gakonst** [comment]
Maybe we should set the standard that remappings in foundry.toml should only be populated for things that you want to override and NOT for dependencies that you access via the default remapping? We don't want the remappings config to evolve into a package.json:dependencies-type thing imo

**Comment by StErMi** [comment]
> Maybe we should set the standard that remappings in foundry.toml should only be populated for things that you want to override and NOT for dependencies that you access via the default remapping? We don't want the remappings config to evolve into a package.json:dependencies-type thing imo

If `forge` handle all the add/remove from the array for me when I install/remove dependencies, there's no problem for me with that solution.

The “only” problem is about the VSCode integration, where you [are saying](https://onbjerg.github.io/foundry-book/guides/vscode.html) to have also the external remapping file

**Comment by onbjerg** [comment]
Perhaps it would be worth to extend the VSCode plugin at this point to accept a remapping command instead so you could specify that you want to use `dapp remappings`/`forge remappings` instead of `remappings.txt`

**Comment by gakonst** [comment]
It's a good idea, although rn `forge remappings` will always output the 'default' detected remappings, and not the ones that have been overridden (via remappings.txt, foundry.toml or env var). So to make the change upstream on [vscode-solidity](https://github.com/juanfranblanco/vscode-solidity/issues/new), we'd prob need that first?

**Comment by onbjerg** [comment]
Yeah, we'd need that. Does `dapp remappings` only output autogenerated ones? If so we could hide this new behavior behind a flag like `--all` or `--with-overrides` or something

**Comment by tfalencar** [comment]
I stumbled on this issue, as I got confused on why `forge remappings` command was picking up my changes in foundry.toml, neither in .dapprc. The doc says the following about this command: "prints the automatically inferred remappings for this repository". Maybe we should either update the documentation, or change the command behaviour? IMO it would be useful if this command actually printed the final result (after reading all config files, env vars) with the effective configuration.

**Comment by tfalencar** [comment]
Hm.. it looks like I was "stuck" in some old forge version, that wasn't even picking up the foundry.toml.

**Comment by onbjerg** [comment]
`forge remappings` docs are accurate - it *only* prints what we automatically infer, not user specified overrides, but I can see how it might be interpreted wrongly. Any suggestions on how to make it clearer?

**Comment by gakonst** [comment]
I guess we could make the default behavior to take into account overrides?

**Comment by onbjerg** [comment]
Would make sense to default include overrides and then have a flag like `--no-overrides` or `--only-inferred` or something

**Comment by tfalencar** [comment]
Maybe some context helps: when I was initially confused about why `remappings` was not picking the overrides, I was in a version of forge where `config` subcommand didn't exist. Today when I see it both commands there (config and remappings), and read the docs, I can infer that since "config" considers the overrides (that is, the "effective" values used), I would guess guess that `remappings` doesn't. So my suggestion would be the following:
Either:
1-  remove remappings altogether
2- clarify exactly what remappings is useful for or how it interacts  with the "local repository"
3- Leave both commands but update documentation with something along the lines of:
`remappings` - prints an example configuration (default values)?
`config` - Print the effective config values.
Relatedly, I didn't find in the forge docs a documentation similar to what we have in dapp-tools on the order of precedence related to config variables, environment vars, etc.

**Comment by tfalencar** [comment]
Maybe some context helps at least where my confusion came from: when I was initially confused about why remappings was not picking the overrides, I was in a version of forge where config subcommand didn't exist. Today when I see both commands there (config and remappings), and read the docs, I can infer that since "config" considers the overrides (that is, the "effective" values used), I would guess guess that remappings doesn't. My suggestion would be the following:
Either:
1- remove remappings subcommand altogether - user can see them anyways with output of config command;
2- do something along the lines of what @onbjerg suggested about with optional flags;
2- clarify exactly what remappings is useful for or how it interacts with the "local repository"  - what affects its output?
3- Leave both commands but update documentation with something along the lines of:
remappings - prints the effective remappings
config - Print all effective configuration values.
Relatedly, I didn't find in the forge docs a documentation similar to what we have in dapp-tools on the order of precedence related to config variables, environment vars, etc. I just tried changing the value of remappings in the foundry.toml config file, yet I didn't see the change reflected in forge config - maybe I'm missing something here, but clarifying the whole topic in general in the docs would be beneficial imo.

**Comment by gakonst** [comment]
> Would make sense to default include overrides and then have a flag like `--no-overrides` or `--only-inferred` or something

Yep. I think this is the right solve here.

**Comment by onbjerg** [comment]
Added to stable milestone considering we ended up only needing to change `forge remappings`

**Comment by lucas-manuel** [comment]
Close-able? @mds1

**Comment by mds1** [comment]
Remappings are automatically inferred and the overrides can be specified in the config file, so agreed with closing
